### ISSUE TITLE
Building pinyin toolkit database on 0.6.4

### ISSUE THREAD

**Issue by Zinbiel** [batterseapower/pinyin-toolkit#180]
I get this error:
File "C:\Users\Jason\Appdata\Roaming.anki\plugins\forms\builddbcontroller.py",
line 54, in run
File "C:\Users\Jason\Appdata\Roaming.anki\plugins\pinyin\db\builder.py", line 52, in build
AttributeError: 'module' object has no attribute 'getDBConnector'

**Comment by batterseapower** [comment]
Apologies, the release had a bug. Please try v0.6.5 (confirmed to work).

**Comment by unsermanninchina** [comment]
hi batterseapower,

have quite the same problem with fresh installed plugin v0.6.5, online installed from Anki 0.9.9.8.6 (on Kubuntu 10.4). For using Eclectus I have installed cjk.lib:
Traceback ... 

File "/home/../.anki/plugins/pinyin/db/builder.py",
line 54, in run
      dbbilder.build()
File "home/../.anki/plugins/pinyin/db/builder.py", line 52,
in build
    database = ckjlib.dbconnector.getDBconnector({ "url" :
sqalchemy.engine.url.URL("sqlite",
database=self.builtdatabasepath)})
AttbributeError:'module' object has no attribute 
'getDBconnector'

Any hints?
Thank you,

Marcus

**Comment by batterseapower** [comment]
Hi Marcus,

Tihs is kind of strange because the code for v0.6.5 says getDBConnector not getDBconnector (note the case of the "C"). Can you try deleting the "pinyin" directory in the plugin folder and installing again?

Cheers,
Max

**Comment by unsermanninchina** [comment]
Hi Max,

I manually wrote the Traceback, it was a "C" not a "c", sorry. But the proble still appears. After deleting all pinyin files from the pluginfolder and reinstalling, Iget the same problem>

2010-07-07 09:44:54,873 - ERROR - Suppressed exception in database build process
Traceback (most recent call last):
  File "/home/mlinux/.anki/plugins/pinyin/forms/builddbcontroller.py", line 54, in run
    dbbuilder.build()
  File "/home/mlinux/.anki/plugins/pinyin/db/builder.py", line 52, in build
    database = cjklib.dbconnector.getDBConnector({ "url" : sqlalchemy.engine.url.URL("sqlite", database=self.builtdatabasepath) })
AttributeError: 'module' object has no attribute 'getDBConnector'
2010-07-07 09:45:16,124 - ERROR - Database construction failed: disabling the Toolkit

I have cjklib, is that the problem?

Thank you,
Marcus

**Comment by batterseapower** [comment]
Hi Marcus,

Having cjklib already could cause the problem. The Toolkit ships with its own copy of the library, which it expects to be using. Currently, I think it will only work with a version released on Feb 15 or later.

If you change the line in pinyin/**init**.py saying this:

sys.path.append(utils.toolkitdir("pinyin", "vendor", vendor_package))

To this:

sys.path = [utils.toolkitdir("pinyin", "vendor", vendor_package)] + sys.path

Does it start working?

**Comment by unsermanninchina** [comment]
Hi Max,

I changed init.py like you said. It started building the db. 

At the end of building the db a popop said, it had a problem setting up Hanzi Graph and asked for installing python-matplotlib. But Anki did show the plugin already. I installed python-matplotlib, python-matplotlib-data and python-pyparsing.

Finally Installed he audio in pinyin-toolkits preferences, restart ANKI - everything works fine!!!
Thank you! Cheers, Marcus
